docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/update-folder-structure.md

```diff
@@ -0,0 +1,356 @@
+---
+title: v4 Plugin Migration - Update folder structure - Strapi Developer Docs
+description:
+canonicalUrl:
+sidebarDepth: 2
+---
+
+<!-- TODO: update SEO -->
+
+# v4 plugin migration: Update the folder structure
+
+!!!include(developer-docs/latest/update-migration-guides/migration-guides/v4/snippets/plugin-migration-intro.md)!!!
+
+::: callout
+Strapi v3 plugins required a specific folder structure.
+
+In Strapi v4, plugins are developed using a programmatic API, which gives flexibility in the folder structure.
+:::
+
+The folder structure of a Strapi v4 plugin should meet 2 requirements:
+
+- The root of the plugin folder should include a `strapi-server.js` and a `strapi-admin.js` entry files, for the [Server API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md) and [Admin Panel API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md), respectively.
+
+- `strapi-admin.js` and `strapi-server.js` should import other files and folders.
```

Comment:
It is not true as all the code could be done in this file. But you can say that : `strapi-admin.js` and `strapi-server.js` should export the correct structure.
(with a link to the corresponding doc I guess)

Reply:
Thank you for the feedback, guys.
We're trying to encourage people to respect best practices in this guide (in all our documentation, actually).  
So, even if technically you could build an entire plugin in `strapi-server` (or `strapi-admin`), would you consider it as a best practice?

I'll also update the wording (and indeed, we use "interface").

Reply:
Updated [in this commit](https://github.com/strapi/documentation/pull/663/commits/1f0857de27f9ba7cb5392d5394a0dc5a0b7fba57). WDYT?